src/include/sof/atomic.h

```diff
@@ -5,6 +5,8 @@
  * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
  */
 
+#ifndef __ZEPHYR__
+
 #ifndef __SOF_ATOMIC_H__
 #define __SOF_ATOMIC_H__
```

Comment:
I think it's usual to keep this repeated-inclusion guard on the outside of any other instructions. So, I'd put the `#ifndef __ZEPHYR__` under them. Further, I don't think it's good to rely on indirect inclusion of Zephyr's sys/atomic.h, I'd rather add an
```
#else
#include <sys/atomic.h>
```
snippet.

Reply:
done